src/AuthorizedSocketConnection.js

```diff
@@ -176,11 +180,16 @@ export default class AuthorizedSocketConnection<TContext, TCredentials> {
         document,
         null,
         {
-          subscribe: async (...args) =>
-            AsyncUtils.filter(
-              await subscriptionContext.subscribe(...args),
-              this.isAuthorized,
-            ),
+          subscribe: async (topic, { hasPermission, ...options } = {}) => {
```

Comment:
```suggestion
          subscribe: async (topic, { hasPermission = this.config.hasPermission, ...options } = {}) => {
```

instead of the `||` below

Reply:
i get this error, any idea why?
![Screen Shot 2019-04-01 at 1 46 26 PM](https://user-images.githubusercontent.com/3424190/55348214-af00eb00-5484-11e9-8d03-b8bfaff5b1ac.png)

Reply:
what's the inferred type for that 2nd options argument? or what's the full error message there?

Reply:
`Cannot call `this.isAuthorized` with `hasPermission` bound to `hasPermission` because  `TCredentials` [1] is incompatible with  `TCredentials` [2] in the second argument.Flow(InferError)`

Reply:
this is a sign that the type inference isn't working – you'd need to explicitly template this on `TCredentials`

Reply:
template what on `TCredentials`? sorry

Reply:
here: https://github.com/4Catalyzer/graphql-subscription-server/pull/37/files/d145f951d7cce20500e76bbce06012f7a4319f58#diff-ff71f0ff01e3f79aaa23ebaac75c714dR54

but as @jquense said, i think it's fine to just use `<any>` instead of `<*>` and leave it for after unblocking the feature